libp2p/wire.nim

```diff
@@ -79,28 +87,46 @@ proc createStreamServer*[T](ma: MultiAddress,
                             child: StreamServer = nil,
                             init: TransportInitCallback = nil): StreamServer =
   ## Create new TCP stream server which bounds to ``ma`` address.
-  var address = initTAddress(ma)
-  if address.family in {AddressFamily.IPv4, AddressFamily.IPv6}:
-    if ma[1].tryGet().protoCode().tryGet() != multiCodec("tcp"):
-      raise newException(TransportAddressError, "Incorrect address type!")
-  result = createStreamServer(address, cbproc, flags, udata, sock, backlog,
+  if not(RTRANSPMA.match(ma)):
+    raise newException(TransportAddressError, "Incorrect address type!")
+
+  let address = initTAddress(ma)
+  result = createStreamServer(address.tryGet(), cbproc, flags, udata, sock,
+                              backlog, bufferSize, child, init)
+
+proc createStreamServer*[T](ma: MultiAddress,
+                            flags: set[ServerFlags] = {},
+                            udata: ref T,
+                            sock: AsyncFD = asyncInvalidSocket,
+                            backlog: int = 100,
+                            bufferSize: int = DefaultStreamBufferSize,
+                            child: StreamServer = nil,
+                            init: TransportInitCallback = nil): StreamServer =
+  ## Create new TCP stream server which bounds to ``ma`` address.
+  if not(RTRANSPMA.match(ma)):
+    raise newException(TransportAddressError, "Incorrect address type!")
```

Comment:
Same problem with exception type and message.

Reply:
Noted, I'll make thin non fatal.